core/sds-aspect-model-validator/src/main/java/io/openmanufacturing/sds/aspectmodel/shacl/RustLikeFormatter.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (c) 2023 Robert Bosch Manufacturing Solutions GmbH
+ *
+ * See the AUTHORS file(s) distributed with this work for additional
+ * information regarding authorship.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package io.openmanufacturing.sds.aspectmodel.shacl;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.apache.jena.graph.AnyNode;
+import org.apache.jena.graph.BlankNode;
+import org.apache.jena.graph.LiteralNode;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.UriNode;
+import org.apache.jena.graph.VariableNode;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.RDFList;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.sparql.graph.NodeConst;
+import org.apache.jena.vocabulary.RDF;
+
+import io.openmanufacturing.sds.aspectmodel.resolver.parser.IRdfTextFormatter;
+import io.openmanufacturing.sds.aspectmodel.resolver.parser.PlainTextFormatter;
+import io.openmanufacturing.sds.aspectmodel.resolver.parser.SmartToken;
+
+/**
+ * Rust-like message formatter. Formatted messages look something like the following example:
+ *
+ * ---> Error at line 11 column 20
+ *    |
+ * 11 |   :testProperty [ a :SomethingElse ]
+ *    |                     ^^^^^^^^^^^^^^ Property ':testProperty' on ':Foo' has type ':SomethingElse', but only ':TestClass2' is allowed.
+ *    |
+ */
+public class RustLikeFormatter {
+
+   private SmartToken highlightToken;
+   private int currentColumn;
+   private final IRdfTextFormatter textFormatter;
+
+   // some statements (like anonymous nodes or lists) can be handled out-of-order, so we remember them to not go over them twice
+   private final Set<Statement> seen = new HashSet<>();
+   private List<Statement> candidateStatements;
+
+   // The parsed model does not contain all the original tokens ( braces in lists, semicolons between statements etc.). But as we want to achieve
+   // as nice and natural formatting as possible, we look at the available information to achieve the proper spacing.
+   private List<Integer> knownPositions;
+
+   public RustLikeFormatter() {
+      textFormatter = new PlainTextFormatter();
```

Comment:
I'd rather call the other constructor here, i.e. `this( new PlainTextFormatter() )`.
Also please add a comment to the constructor that explains that it will default to a PlainTextFormatter.